tests/pkg/k8s/jobs.go

```diff
@@ -131,6 +131,24 @@ func WaitForJob(t *testing.T, ctx context.Context, k8sClient *kubernetes.Clients
 	}
 }
 
+func WaitForJobToFinish(t *testing.T, ctx context.Context, k8sClient *kubernetes.Clientset, jobName, namespace string) {
```

Comment:
In order to avoid code duplication you can make a make function reusable by returning the job status and to reuse it in WaitForJob with desired state.